common/attest_plugin.h

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/internal/hexdump.h>
+#include <openenclave/internal/print.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/report.h>
+#include <openenclave/internal/safemath.h>
+#include <openenclave/internal/utils.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "common.h"
+
+#include <openenclave/internal/plugin.h>
+
+/**
+ * Evidence header: the structure that the OE SDK runtime puts on top of
+ * evidence data.
+ */
+typedef struct _oe_attestation_header
+{
+    /* Set to OE_ATTESTATION_HEADER_VERSION. */
+    uint32_t version;
+
+    /* UUID to identify format. */
+    oe_uuid_t format_id;
+
+    /* Size of evidence/endorsements sent to the plugin. */
+    uint64_t data_size;
+
+    /* The actual data */
+    uint8_t data[];
+
+    /* data_size bytes that follows the header will be sent to a plugin. */
```

Comment:
This format requires data copying, as opposed to just using a pointer to the actual data.

Reply:
As discussed in today's SIG-attestation meeting, this header structure is part of the existing implementation. Will be addressed as part of PR #2999